litmus-portal/frontend/src/views/MyHub/MyHubConnect/index.tsx

```diff
@@ -0,0 +1,226 @@
+                      GitURL: gitHub.GitURL,
+                      GitBranch: gitHub.GitBranch,
+                    })
+                  }
+                />
+              </div>
+              <div className={classes.inputField}>
+                <InputField
+                  label="Git URL"
+                  value={gitHub.GitURL}
+                  helperText={
+                    !isValidWebUrl(gitHub.GitURL) ? 'Enter a valid URL' : ''
+                  }
+                  validationError={!isValidWebUrl(gitHub.GitURL)}
+                  required
+                  handleChange={(e) =>
+                    setGitHub({
+                      HubName: gitHub.HubName,
+                      GitURL: e.target.value,
+                      GitBranch: gitHub.GitBranch,
+                    })
+                  }
+                />
+              </div>
+              <div className={classes.inputField}>
+                <InputField
+                  label="Branch"
+                  value={gitHub.GitBranch}
+                  helperText={
+                    validateStartEmptySpacing(gitHub.GitBranch)
+                      ? 'Should not start with an empty space'
+                      : ''
+                  }
+                  validationError={validateStartEmptySpacing(gitHub.GitBranch)}
+                  required
+                  handleChange={(e) =>
+                    setGitHub({
+                      HubName: gitHub.HubName,
+                      GitURL: gitHub.GitURL,
+                      GitBranch: e.target.value,
+                    })
+                  }
+                />
+              </div>
+              <div className={classes.submitBtnDiv}>
+                <ButtonFilled isPrimary={false} type="submit">
+                  {t('myhub.connectHubPage.submitBtn')}
+                </ButtonFilled>
+              </div>
+              <Unimodal open={isOpen} handleClose={handleClose} hasCloseBtn>
+                <div className={classes.modalDiv}>
+                  {cloningRepo ? (
+                    <div>
+                      <Loader />
+                      <Typography className={classes.modalDesc}>
+                        {t('myhub.connectHubPage.cloningText')}
+                      </Typography>
+                    </div>
+                  ) : (
+                    <div>
+                      {error.length ? (
+                        <div>
+                          <Typography
+                            gutterBottom
+                            className={classes.modalHeading}
+                          >
+                            <strong>
+                              {t('myhub.connectHubPage.errorText')}
+                            </strong>{' '}
+                            {t('myhub.connectHubPage.cretingHub')}
+                          </Typography>
+                          <Typography className={classes.modalDesc}>
+                            Error: {error}
```

Comment:
Might feel redundant as it's only one word, see if this is worth including into the translation file as well.

Reply:
@A4k9  I have added the error text in the translation file. This `Error: {error}` displays the error message that we get from the server.